tests/unit/CommandTest.php

```diff
@@ -1455,12 +1455,14 @@ public function testCreateView()
             ->select('bar')
             ->from('testCreateViewTable')
             ->where(['>', 'bar', '5']);
-        if ($db->getSchema()->getTableSchema('testCreateViewTable')) {
-            $db->createCommand()->dropTable('testCreateViewTable')->execute();
-        }
+        // Order of the next two "if" conditions is very important, because testCreateView depends on testCreateViewTable.
+        // If testCreateViewTable was deleted before testCreateView, at the second round this test will always fail becase testCreateView exists but it is referred to not existing table
         if ($db->getSchema()->getTableSchema('testCreateView') !== null) {
             $db->createCommand()->dropView('testCreateView')->execute();
         }
+        if ($db->getSchema()->getTableSchema('testCreateViewTable')) {
+            $db->createCommand()->dropTable('testCreateViewTable')->execute();
```

Comment:
Just like for float casting, I'd like to understand why is this change needed if it used to work on yii2.

Reply:
I was saying: If this is a test copied from yii2, and it used to work with yii2, but not anymore with yii3, the tests should not be changed. 

Instead, we should look for the reason it's failing now, cause this may be caused by a new bug.

Can you check what's different between yii2 and yii3 here?

Reply:
Surely, when Travis starts the test, its database not exists, so this code change is unuseful.

It is useful if you start from existing database (if you have already executed the test), because the testCreateViewTable is not dropped by mysql.sql.

Try.

Reply:
This is what happens using previous code version:

```bash
root@ecf9af83c6f8:/app/web/db-mysql# echo "create database yiitest" | mysql -h mysql -u travis -ptravis
root@ecf9af83c6f8:/app/web/db-mysql# vendor/bin/phpunit tests/unit/CommandTest.php --filter testCreateView
PHPUnit 7.5.1 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.3 with Xdebug 2.6.0-dev
Configuration: /app/web/db-mysql/phpunit.xml.dist

.                                                                   1 / 1 (100%)

Time: 1.4 seconds, Memory: 4.00MB

OK (1 test, 1 assertion)
root@ecf9af83c6f8:/app/web/db-mysql# vendor/bin/phpunit tests/unit/CommandTest.php --filter testCreateView
PHPUnit 7.5.1 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.3 with Xdebug 2.6.0-dev
Configuration: /app/web/db-mysql/phpunit.xml.dist

E                                                                   1 / 1 (100%)

Time: 1.56 seconds, Memory: 4.00MB

There was 1 error:

1) yii\db\mysql\tests\CommandTest::testCreateView
yii\db\Exception: SQLSTATE[HY000]: General error: 1356 View 'yiitest.testCreateView' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them
The SQL being executed was: SHOW FULL COLUMNS FROM `testCreateView`

/app/web/db-mysql/vendor/yiisoft/db/src/Schema.php:669
/app/web/db-mysql/vendor/yiisoft/db/src/Command.php:1263
/app/web/db-mysql/vendor/yiisoft/db/src/Command.php:1148
/app/web/db-mysql/vendor/yiisoft/db/src/Command.php:399
/app/web/db-mysql/src/Schema.php:313
/app/web/db-mysql/src/Schema.php:126
/app/web/db-mysql/vendor/yiisoft/db/src/Schema.php:749
/app/web/db-mysql/vendor/yiisoft/db/src/Schema.php:199
/app/web/db-mysql/vendor/yiisoft/db/tests/unit/CommandTest.php:1463
/app/web/db-mysql/vendor/phpunit/phpunit/phpunit:61

Caused by
PDOException: SQLSTATE[HY000]: General error: 1356 View 'yiitest.testCreateView' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them

/app/web/db-mysql/vendor/yiisoft/db/src/Command.php:1258
/app/web/db-mysql/vendor/yiisoft/db/src/Command.php:1148
/app/web/db-mysql/vendor/yiisoft/db/src/Command.php:399
/app/web/db-mysql/src/Schema.php:313
/app/web/db-mysql/src/Schema.php:126
/app/web/db-mysql/vendor/yiisoft/db/src/Schema.php:749
/app/web/db-mysql/vendor/yiisoft/db/src/Schema.php:199
/app/web/db-mysql/vendor/yiisoft/db/tests/unit/CommandTest.php:1463
/app/web/db-mysql/vendor/phpunit/phpunit/phpunit:61

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

As you see, when at second time I try to launch the same test (with existing database), test will fail

Reply:
What if you pass `true` as second argument of `getTableSchema()`?

Reply:
Then I would remove comment above or change it to `View must be deleted before deleting the tables on which it depends` - right it creates more confusion than lack of it :P

Reply:
Thank you for the feedback!